src/panels/config/integrations/ha-integration-card.ts

```diff
@@ -716,6 +724,69 @@ export class HaIntegrationCard extends LitElement {
         ),
       });
     }
+    if (applicationCredentialsId) {
+      this._removeApplicationCredential(applicationCredentialsId);
+    }
+  }
+
+  // Return an application credentials id for this config entry to prompt the
+  // user for removal. This is best effort so we don't stop overall removal
+  // if the integration isn't loaded or there is some other error.
+  private async _applicationCredentialForRemove(entryId: string) {
+    try {
+      return (await fetchApplicationCredentialsConfigEntry(this.hass, entryId))
```

Comment:
will this always return an object?

Reply:
I am not sure this is answering what you're looking for, but my impression is this will either:
- return a result with the `application_credentials_id` field when there are creds
- return an empty result in the common case where there are no creds, so `application_credentials_id` will not be in the response
- send an error of `invalid_config_entry_id`

My intent was that case 2 above would return undefined.

Let me know if there is a better pattern to follow.

Reply:
If case 2 returns undefined/null, we should make this:
```ts
      return (await fetchApplicationCredentialsConfigEntry(this.hass, entryId))?.application_credentials_id;
```

and update the typing

Reply:
I was thinking that case 2 returns an empty message, so just the `application_credentials_id` field is undefined set:

```
export interface ApplicationCredentialsConfigEntry {
  application_credentials_id?: string;
}
```

Am i understanding that right?

Reply:
ok, in that case it is fine 👍